File: testdata/fixtures/bench/zig/vulnerable/a03_command_injection.zig
// Source: CWE-78 - OS Command Injection via ChildProcess in Zig
// Expected: BATOU-ZIG
// OWASP: A03:2021 - Injection (Command Injection)

const std = @import("std");

pub fn runCommand(user_input: []const u8) !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = arena.allocator();

    const cmd = try std.fmt.allocPrint(allocator, "ping -c 1 {s}", .{user_input});
    var child = std.process.Child.init(.{
        .argv = &[_][]const u8{ "/bin/sh", "-c", cmd },
    }, allocator);
    _ = try child.spawnAndWait();
}

pub fn main() !void {
    var args = try std.process.argsAlloc(std.heap.page_allocator);
    defer std.process.argsFree(std.heap.page_allocator, args);
    if (args.len > 1) {
        try runCommand(args[1]);
    }
}

File: testdata/fixtures/bench/rust/vulnerable/a03_command_injection.rs
// Source: CWE-78 - OS Command Injection via Command::new in Rust
// Expected: BATOU-RS
// OWASP: A03:2021 - Injection (Command Injection)

use std::env;
use std::process::Command;

fn run_ping(target: &str) {
    let output = Command::new("sh")
        .arg("-c")
        .arg(format!("ping -c 1 {}", target))
        .output()
        .expect("failed to execute");
    println!("{}", String::from_utf8_lossy(&output.stdout));
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() > 1 {
        run_ping(&args[1]);
    }
}
